src/deadline/job_attachments/api/attachment.py

```diff
@@ -28,56 +33,156 @@ def attachment_download(
+):
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+
+    API to download job attachments based on given list of manifests.
+    If path mapping rules file is given, map to corresponding destinations.
+
+    Args:
+        manifests (List[str]): File Path to the manifest file for upload.
+        s3_root_uri (str): S3 root uri including bucket name and root prefix.
+        boto3_session (boto3.Session): Boto3 session for interacting with customer s3.
+        root_dirs (List[str]): List of root directories holding attachments. Defaults to empty.
+        path_mapping_rules (Optional[str], optional): Optional file path to a JSON file contains list of path mapping. Defaults to None.
+        upload_manifest_path (Optional[str], optional): Optional prefix for uploading given manifests. Defaults to None.
+        logger (ClickLogger, optional): Logger to provide visibility. Defaults to ClickLogger(False).
+
+    Raises:
+        NonValidInputError: _description_
+    """
+
+    if not all(os.path.isdir(root) for root in root_dirs):
+        raise NonValidInputError(f"Specified root dir {root_dirs} contains nonvalid directory.")
+
+    file_name_manifest_dict: Dict[str, BaseAssetManifest] = _get_file_name_manifest_dict(
+        manifests=manifests
+    )
+    path_mapping_rule_list: List[PathMappingRule] = _process_path_mapping(
+        path_mapping_rules=path_mapping_rules, root_dirs=root_dirs
+    )
+
+    s3_settings: JobAttachmentS3Settings = JobAttachmentS3Settings.from_s3_root_uri(s3_root_uri)
+    asset_uploader: S3AssetUploader = S3AssetUploader(session=boto3_session)
+    for file_name in file_name_manifest_dict:
+        manifest: BaseAssetManifest = file_name_manifest_dict[file_name]
+
+        # File name is supposed to be prefixed by a hash of source path in path mapping or provided root dirs
+        rule: Optional[PathMappingRule] = next(
+            # search in path mapping to determine source and destination
+            (
+                rule
+                for rule in path_mapping_rule_list
+                if rule.get_hashed_source_path(manifest.get_default_hash_alg()) in file_name
+            ),
+            None,
+        )
+        if not rule:
+            raise NonValidInputError(
+                f"No valid root defined for given manifest {file_name}, please check input root dirs and path mapping rule."
+            )
+
+        # Uploads all files to a CAS in the manifest, optionally upload manifest file
+        key, data = asset_uploader.upload_assets(
+            job_attachment_settings=s3_settings,
+            manifest=manifest,
+            partial_manifest_prefix=upload_manifest_path,
+            source_root=Path(rule.source_path),
+            asset_root=Path(rule.destination_path),
+        )
+        logger.echo(
+            f"Uploaded assets from {rule.source_path}, to {s3_settings.to_s3_root_uri()} key {key}, hashed data {data}"
```

Comment:
```suggestion
            f"Uploaded assets from {rule.source_path}, to {s3_settings.to_s3_root_uri()}/{key}, hashed data {data}"
```

nit: technically the `{key}` part is the last part of the key. From the PR description, the `s3_settings.to_s3_root_uri` returns an S3 bucket + key prefix. I think we can just remove the `key` part from the message and just directly combine `{s3_settings.to_s3_root_uri()}/{key}` to log the full S3 URI. 

What do you think?